drivers/rptun/rptun_ping.c

```diff
@@ -0,0 +1,192 @@
+/****************************************************************************
+ * drivers/rptun/rptun_ping.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <nuttx/arch.h>
+
+#include <inttypes.h>
+#include <string.h>
+
+#include "rptun.h"
+
+/****************************************************************************
+ * Pre-processor definitions
+ ****************************************************************************/
+
+#ifndef MIN
+  #define MIN(n,m)                  (((n) < (m)) ? (n) : (m))
+#endif
+
+#ifndef MAX
+  #define MAX(n,m)                  (((n) < (m)) ? (m) : (n))
+#endif
+
+#define RPTUN_PING_EPT_NAME         "rpmsg-ping"
+#define RPTUN_PING_SEND             1
+#define RPTUN_PING_SEND_NOACK       2
+#define RPTUN_PING_ACK              3
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+begin_packed_struct struct rptun_ping_msg_s
+{
+  uint32_t cmd;
+  uint32_t len;
+  uint64_t cookie;
+} end_packed_struct;
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static int rptun_ping_ept_cb(FAR struct rpmsg_endpoint *ept,
+                             FAR void *data, size_t len, uint32_t src,
+                             FAR void *priv)
+{
+  FAR struct rptun_ping_msg_s *msg = data;
+  FAR sem_t *sem = (FAR sem_t *)(uintptr_t)msg->cookie;
+
+  if (msg->cmd == RPTUN_PING_SEND)
+    {
+      msg->cmd = RPTUN_PING_ACK;
+      rpmsg_send(ept, msg, len);
+    }
+  else if (msg->cmd == RPTUN_PING_ACK)
+    {
+      nxsem_post(sem);
+    }
+
+  return 0;
```

Comment:
Should we return 0 or result of `rpmsg_send` / `nxsem_post`?

Reply:
suggest, don't do that, because the caller of ept_cb don't welcome the error returned.